webapp/tests/karma/ts/services/enketo-prepopulation-data.service.spec.ts

```diff
@@ -135,129 +138,109 @@ describe('EnketoPrepopulationData service', function() {
+    return service
+      .get(editPersonFormWithoutInputs, data)
+      .then((actual) => {
         const xml = $($.parseXML(actual));
-        chai.expect(xml.find('data > person > last_name')[0].innerHTML).to.equal(data.person.last_name);
-        chai.expect(UserSettings.callCount).to.equal(1);
-        done();
-      })
-      .catch(done);
-    rootScope.$digest();
+        expect(xml.find('data > person > last_name')[0].innerHTML).to.equal(data.person.last_name);
+        expect(UserSettings.callCount).to.equal(1);
+      });
   });
 
-  it('binds form content into generated form model', function(done) {
+  it('binds form content into generated form model', () => {
     const data = { person: { name: 'sally' } };
     const user = { name: 'geoff' };
-    UserSettings.returns(Promise.resolve(user));
-    service(generatedForm, data)
-      .then(function(actual) {
+    UserSettings.resolves(user);
+    return service
+      .get(generatedForm, data)
+      .then((actual) => {
         const xml = $($.parseXML(actual));
-        chai.expect(xml.find('data > person > name')[0].innerHTML).to.equal(data.person.name);
-        chai.expect(UserSettings.callCount).to.equal(1);
-        done();
-      })
-      .catch(done);
-    rootScope.$digest();
+        expect(xml.find('data > person > name')[0].innerHTML).to.equal(data.person.name);
+        expect(UserSettings.callCount).to.equal(1);
+      });
   });
 
-  it('binds user details and form content into model', function(done) {
+  it('binds user details and form content into model', () => {
     const data = { person: { last_name: 'salmon' } };
     const user = { name: 'geoff' };
-    UserSettings.returns(Promise.resolve(user));
-    service(editPersonForm, data)
-      .then(function(actual) {
+    UserSettings.resolves(user);
+    return service
+      .get(editPersonForm, data)
+      .then((actual) => {
         const xml = $($.parseXML(actual));
-        chai.expect(xml.find('inputs > user > name')[0].innerHTML).to.equal(user.name);
-        chai.expect(xml.find('data > person > last_name')[0].innerHTML).to.equal(data.person.last_name);
-        chai.expect(UserSettings.callCount).to.equal(1);
-        done();
-      })
-      .catch(done);
-    rootScope.$digest();
+        expect(xml.find('inputs > user > name')[0].innerHTML).to.equal(user.name);
+        expect(xml.find('data > person > last_name')[0].innerHTML).to.equal(data.person.last_name);
+        expect(UserSettings.callCount).to.equal(1);
+      });
   });
 
-  it('binds form content into model with custom root node', function(done) {
+  it('binds form content into model with custom root node', () => {
     const data = { person: { last_name: 'salmon' } };
     const user = { name: 'geoff' };
-    UserSettings.returns(Promise.resolve(user));
-    service(pregnancyForm, data)
-      .then(function(actual) {
+    UserSettings.resolves(user);
+    return service
+      .get(pregnancyForm, data)
+      .then((actual) => {
```

Comment:
What are your thoughts about asserting false on the catch when we return promises in unit test? We do this in some places but not very consistent 🤔 
We can do this in critical places or always as good practice. For the later we could create a ToDo to go back and add the assertion.

Reply:
Unless we're expecting the code to throw an error, I see no point in asserting in `catch` (it's just bloat). 
Can you link me to a place where we assert in `catch` when we don't expect an error? 

If the code throws and there's no catch, Jasmine/Mocha/whatever will report the test as failed. 
I think we should only assert in `catch` when we expect an error to be thrown, and in these cases, we should always throw an error in the `then`.